zendesk/user.go

```diff
@@ -135,4 +135,25 @@ func (z *Client) CreateUser(ctx context.Context, user User) (User, error) {
 	return result.User, nil
 }
 
+// CreateOrUpdateUser creates new user
+// ref: https://developer.zendesk.com/rest_api/docs/core/triggers#create-trigger
+func (z *Client) CreateOrUpdateUser(ctx context.Context, user User) (User, error) {
```

Comment:
Probably should add a test for this method.

Check this file for examples: https://github.com/nukosuke/go-zendesk/blob/master/zendesk/user_test.go

Reply:
Added a test case, but pretty naive and simple using the existing create_user test. I don't have much xp with the fixture driven tests so I may be missing things.